### DISCUSSION TITLE
initialChain defaulting to currently connected chain if supported

### DISCUSSION THREAD

**Discussion by 0xModene** [rainbow-me/rainbowkit#633] [Ideas]
Current behavior for `initialChain` is that if left empty, it defaults to the first chain in provided to `RainbowKitProvider`. 

This should probably default to whatever chain the user is currently connected to, if supported, then fall back to the first chain in `chains`

**Comment by spaceturtleship** [comment]
Yea, either this, accepting a list of supported chains, or optionally disabling the feature would be useful. If an app supports multiple chains, and the user is already on a chain that is supported, it doesn't make sense to prompt them to change.

Also if they decline to change, they are already connected but RainbowKit displays the "Error connecting, please retry!" message which is kind of confusing.

**Comment by markdalgleish** [comment]
I agree this would be much better and I've been wanting to switch to this for a while. This feature requires a change to wagmi so I've opened a draft PR there to get the ball rolling: https://github.com/wagmi-dev/wagmi/pull/828

**Comment by markdalgleish** [comment]
Turns out there is an API that lets us handle this ourselves (thanks @jxom 🙏 ) https://github.com/wagmi-dev/wagmi/pull/828#issuecomment-1211382281

**Comment by markdalgleish** [comment]
This is now the default behaviour in [RainbowKit v0.4.6](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.6).

**Reply by 0xModene** [reply]
absolute 👑 thanks Mark! Just reviewed the code myself, and I'd come up with a similar solution on my own I was prepared to PR but you knocked that out of the park. Merging into Index Coop app now
